### PR TITLE
Add config switch for basic auth

### PR DESCRIPTION
**Pull request by spacemanspiff2007**
First part of #1177

### CODE REVIEW

**Comment by spacemanspiff2007** [comment]
I did not sign off because imho this is only a single doc change

**Comment by spacemanspiff2007** [comment]
@kaikreuzer 
I've updated this pr with your expire parameter.
Do you think this can be merged?

File: distributions/openhab/src/main/resources/conf/services/runtime.cfg [hunk 1]
```diff
@@ -77,3 +77,15 @@ org.openhab.ephemeris:dayset-school=[MONDAY,TUESDAY,WEDNESDAY,THURSDAY,FRIDAY]
 # so that they are immediately available in the system (default is false)
 #
 #org.openhab.inbox:autoApprove=true
+
+################ REST API ###################
+
+# Setting this to true will allow basic authentification for the rest api
```
Comment (kaikreuzer) [inline distributions/openhab/src/main/resources/conf/services/runtime.cfg:83]: Could you also use the same description as in https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.io.rest.auth/src/main/resources/OH-INF/config/config.xml#L11-L12?

**Review by kaikreuzer — COMMENTED** [review]
Thanks, looks good.
May I ask you to also document the [third parameter](https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.io.rest.auth/src/main/resources/OH-INF/config/config.xml#L26-L34), so that it is complete here?

**Comment by spacemanspiff2007** [comment]
I've added it accordingly

File: distributions/openhab/src/main/resources/conf/services/runtime.cfg [hunk 2]
```diff
@@ -77,3 +77,21 @@ org.openhab.ephemeris:dayset-school=[MONDAY,TUESDAY,WEDNESDAY,THURSDAY,FRIDAY]
...
+# authorization. The entries in the cache expire after a while in order to not keep credentials in memory indefinitely.
+# This value defines the expiration time in hours. Set it to 0 for disabling the cache.
+#
+#org.openhab.restauth:cacheExpiration=6
+
+# By default, operations requiring the "user" role are available when unauthenticated.
+# Disabling this option will enforce authorization for these operations.
+# Warning: This causes clients that do not support authentication to break.
+#
+#org.openhab.restauth:implicitUserRole=False
```
Comment (kaikreuzer) [inline distributions/openhab/src/main/resources/conf/services/runtime.cfg:97]:
```suggestion
#org.openhab.restauth:implicitUserRole=false
```

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**